scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/ScriptData.java

```diff
@@ -103,29 +103,68 @@ static ScriptData createForFlowScript(FlowScript script, TaskData taskData) {
         return scriptData;
     }
 
-    FlowScript createFlowScript() throws InvalidScriptException {
-        if (flowScriptActionType == null) {
-            throw new DatabaseManagerException("Flow script action type is null");
-        }
-
+    FlowScript createFlowScriptByScript() throws InvalidScriptException {
         if (flowScriptActionType.equals(FlowActionType.CONTINUE.toString())) {
             return FlowScript.createContinueFlowScript();
         } else if (flowScriptActionType.equals(FlowActionType.IF.toString())) {
             return FlowScript.createIfFlowScript(getScript(),
                                                  getScriptEngine(),
                                                  getFlowScriptTarget(),
                                                  getFlowScriptTargetElse(),
-                                                 getFlowScriptTargetContinuation());
+                                                 getFlowScriptTargetContinuation(),
+                                                 parameters());
         } else if (flowScriptActionType.equals(FlowActionType.LOOP.toString())) {
-            return FlowScript.createLoopFlowScript(getScript(), getScriptEngine(), getFlowScriptTarget());
+            return FlowScript.createLoopFlowScript(getScript(), getScriptEngine(), getFlowScriptTarget(), parameters());
         }
         if (flowScriptActionType.equals(FlowActionType.REPLICATE.toString())) {
-            return FlowScript.createReplicateFlowScript(getScript(), getScriptEngine());
+            return FlowScript.createReplicateFlowScript(getScript(), getScriptEngine(), parameters());
         } else {
             throw new DatabaseManagerException("Invalid flow script action: " + flowScriptActionType);
         }
     }
 
+    FlowScript createFlowScriptByURL() throws InvalidScriptException {
+        try {
+            URL inputUrl = new URL(url);
```

Comment:
maybe initialize the URL outside the try block

Reply:
Actually the try block is there just because of the exception that can be thrown by the URL constructor. :p

Reply:
intialize an URL can throw a checked exception so it makes sense to include it inside the try block (to allow throwing an InvalidScriptEception)